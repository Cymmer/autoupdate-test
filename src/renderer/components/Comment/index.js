import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import { textTypes } from 'components/elements/constants';
import { Text, UserImage } from 'components/elements';
import { formatTeacherName } from 'codechum-app-utils';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

const Comment = ({ user, userType, time, content }) => (
  <div className={styles.Comment}>
    <UserImage
      image={user.profile_pic}
      imageClassName={styles.Comment_userImage}
    />
    <div className={styles.Comment_info}>
      <div
        className={cn({
          [styles.Comment_user]: userType === GLOBALS.USER_TYPES.STUDENT,
        })}
      >
        <Text
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['700']}
          type={textTypes.DATA.XS}
        >
          {formatTeacherName(user)}
        </Text>
        <Text
          className={cn({
            [styles.Comment_time___student]:
              userType === GLOBALS.USER_TYPES.STUDENT,
            [styles.Comment_time___teacher]:
              userType === GLOBALS.USER_TYPES.TEACHER,
          })}
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
          type={textTypes.BODY.XS}
        >
          {time}
        </Text>
      </div>
      <div
        className={cn({
          [styles.Comment_content]: userType === GLOBALS.USER_TYPES.STUDENT,
        })}
      >
        <Text
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['700']}
          type={textTypes.BODY.XS}
        >
          {content}
        </Text>
      </div>
    </div>
  </div>
);

Comment.propTypes = {
  user: PropTypes.shape({
    profile_pic: PropTypes.string,
    first_name: PropTypes.string,
    last_name: PropTypes.string,
  }).isRequired,
  userType: PropTypes.oneOf([
    GLOBALS.USER_TYPES.STUDENT,
    GLOBALS.USER_TYPES.TEACHER,
  ]).isRequired,
  time: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
};

export default Comment;
