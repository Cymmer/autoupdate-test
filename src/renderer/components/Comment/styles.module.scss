@import '../../scss/variables';
@import '../../scss/mixins';
@import '../../scss/themes';

@mixin link($color) {
  @include heading(0.75rem, 1.67);
  color: $color;
  text-decoration: none;
  padding: 6px 0;

  &:hover {
    color: darken($color, 10%);
  }
}

.Comment {
  display: flex;
  margin: 0 0 24px 0;

  &_userImage {
    width: 36px;
    margin: 0 12px 0 0;
  }

  &_time {
    &___teacher {
      margin: 0 0 8px 0;
    }

    &___student {
      margin: 0 0 0 8px;
    }
  }

  &_user {
    display: flex;
    align-items: center;
  }

  &_info {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
  }

  &_content {
    margin: 4px 0 0 0;
    padding: 6px 16px;
    border-radius: 16px;
    width: auto;
    border: 1px solid;
    @include theme {
      background-color: t('cardBackgroundColor');
      border-color: t('borderColor');
    }
  }
}
