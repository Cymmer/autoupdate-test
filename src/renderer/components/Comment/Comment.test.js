import React from 'react';
import { shallow } from 'enzyme';
import Comment from './index';

const COMPONENT_PROPS = {
  time: '10 hours ago',
  user: {
    first_name: 'Male',
    last_name: 'Teacher',
    profile_pic: 'http://example.com',
  },
  content: 'Good job',
};

describe('RENDERING - Comment', () => {
  it('should render', () => {
    const component = shallow(<Comment {...COMPONENT_PROPS} />);
    expect(component).toMatchSnapshot();
  });
});
