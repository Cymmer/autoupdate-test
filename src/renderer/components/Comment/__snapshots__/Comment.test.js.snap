// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Comment should render 1`] = `
<div
  className="Comment"
>
  <UserImage
    className={null}
    id={null}
    image="http://example.com"
    imageClassName="Comment_userImage"
  />
  <div
    className="Comment_info"
  >
    <div
      className=""
    >
      <Text
        className={null}
        colorClass="n___700"
        element={null}
        id={null}
        type="data___xs"
      >
        Male Teacher
      </Text>
      <Text
        className=""
        colorClass="n___400"
        element={null}
        id={null}
        type="body___xs"
      >
        10 hours ago
      </Text>
    </div>
    <div
      className=""
    >
      <Text
        className={null}
        colorClass="n___700"
        element={null}
        id={null}
        type="body___xs"
      >
        Good job
      </Text>
    </div>
  </div>
</div>
`;
